File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxArea=0;
        
        int left=0, right=height.size()-1;

        while(left < right)
        {
            int width=right - left;
            maxArea=max(maxArea,min(height[left],height[right]) * width);

            if(height[left] <= height[right])
                left++;
            else
                right--;
        }

        return maxArea;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        
        int water = 0;
        int i = 0, j = n - 1;
        
        while(i < j) {
            int w = j - i;
            int h = Math.min(height[i], height[j]);
            
            water = Math.max(water, w * h);
            
            if(height[i] < height[j]) {
                i++;
            } else {
                j--;
            }
        }
        
        return water;
    }
}